libs/langchain-core/src/messages/base.ts

```diff
@@ -546,8 +546,16 @@ export function _mergeObj<T = any>(
     return (left + right) as T;
   } else if (Array.isArray(left) && Array.isArray(right)) {
     return _mergeLists(left, right) as T;
-  } else if (typeof left === "object" && typeof right === "object") {
-    return _mergeDicts(left, right) as T;
+  } else if (
+    typeof left === "object" &&
+    typeof right === "object" &&
+    left !== null &&
+    right !== null
+  ) {
```

Comment:
_mergeDicts handles null inputs
```suggestion
  } else if (typeof left === "object" && typeof right === "object") {
```